_overviews/scala3-book/ca-type-classes.md

```diff
@@ -20,99 +20,171 @@ A type class is useful in multiple use-cases, for example:
 - Expressing how a type you don’t own---from the standard library or a third-party library---conforms to such behavior
 - Expressing such a behavior for multiple types without involving sub-typing relationships between those types
 
-In Scala 3, type classes are just traits with one or more parameters whose implementations are provided by `given` instances.
-
-
+Type classes are traits with one or more parameters whose implementations are provided as `given` instances in Scala 3 or `implicit` values in Scala 2.
 
 ## Example
 
-For example, `Show` is a well-known type class in Haskell, and the following code shows one way to implement it in Scala 3.
-If you imagine that Scala classes don’t have a `toString` method, you can define a `Show` type class to add this behavior to any class that you want to be able to convert to a custom string.
+For example, `Show` is a well-known type class in Haskell, and the following code shows one way to implement it in Scala.
+If you imagine that Scala classes don’t have a `toString` method, you can define a `Show` type class to add this behavior to any type that you want to be able to convert to a custom string.
 
 ### The type class
 
 The first step in creating a type class is to declare a parameterized trait that has one or more abstract methods.
 Because `Showable` only has one method named `show`, it’s written like this:
 
+{% tabs 'definition' class=tabs-scala-version %}
+{% tab 'Scala 2' %}
+```scala
+// a type class
+trait Showable[A] {
+  def show(a: A): String
+}
+```
+{% endtab %}
+{% tab 'Scala 3' %}
 ```scala
 // a type class
 trait Showable[A]:
   extension(a: A) def show: String
 ```
+{% endtab %}
+{% endtabs %}
 
 This is the Scala 3 way of saying that any type that implements this trait must define how the `show` method works.
 Notice that the syntax is very close to a normal trait:
```

Comment:
I don't like this line either, it **is** a normal trait - but that can be a follow up PR

Reply:
There is an important distinction that needs to be made on “classic” OOP design vs “type-class-based” design. I will improve the wording.

Reply:
I’ve reworded in 92c1f63f0e5aad94bf27eb261ef2cbcd305ef837